mod/web_services/lib/web_services.php

```diff
@@ -147,19 +141,45 @@ function get_parameters_for_method($method) {
 
 	$sanitised = array();
 
-	// if there are parameters, sanitize them
-	if (isset($API_METHODS[$method]['parameters'])) {
-		foreach ($API_METHODS[$method]['parameters'] as $k => $v) {
-			$param = get_input($k); // Make things go through the sanitiser
-			if ($param !== '' && $param !== null) {
-				$sanitised[$k] = $param;
-			} else {
-				// parameter wasn't passed so check for default
-				if (isset($v['default'])) {
-					$sanitised[$k] = $v['default'];
-				}
-			}
+	$expected_parameters = $API_METHODS[$method]['parameters'];
+
+	if (empty($expected_parameters) || !is_array($expected_parameters)) {
+		// This method has no expected parameters
+		return $sanitised;
+	}
+
+	foreach ($expected_parameters as $key => $spec) {
+		// Make things go through the sanitiser
+		$default = elgg_extract('default', $spec);
+		$value = get_input($key, $default);
+
+		// Cast values to specifid type
+		$type = elgg_extract('type', $spec);
+		switch ($type) {
```

Comment:
Yes, sounds nice. Just need to normalize types. I think we need to add type normalization to ```elgg_ws_expose_function()```